core/unix/memquery.c

```diff
@@ -143,17 +143,20 @@ memquery_library_bounds_by_iterator(const char *name, app_pc *start /*IN/OUT*/,
                         ASSERT_CURIOSITY((slash - src) < dstsz);
                         /* we keep the last '/' at end */
                         ++slash;
-                        strncpy(dst, src, MIN(dstsz, (slash - src)));
+                        int copy_bytes = MIN(dstsz - 1, slash - src);
+                        strncpy(dst, src, copy_bytes);
+                        dst[copy_bytes] = '\0';
```

Comment:
I would think this should be `dst[copy_bytes -  1]`, right?

Reply:
Note that copy_bytes is either equal to the length of the full directory path, or `dstsz - 1` (pretend the **- 1** is bolded).

So, we know definitively that after the `strncpy(dst, src, copy_bytes)`, `dst` is not 0-terminated; quoting the manpage:

    Warning: If there is no null byte among the first n bytes of src, the string placed in dest will not be null-terminated.

Since `copy_bytes` is definitely less than the index of the 0 byte in the `src` string, there will be no 0 byte written in the first `copy_bytes` of `dst`. We want to 0-terminate the string immediately after the `copy_bytes` of data, so `dst[copy_bytes]` should be the correct index unless I'm seriously off-by-one-ing myself here.

(my base-case is: if `dstsz == 1`, we should indeed be strncpy-ing 0 bytes (i.e. `copy_bytes == 0`) and should be setting `dst[0] = '\0'`.)

Reply:
I see, the -1 in the capacity case is up above.  Please add a brief comment on or very near line 148, b/c just looking at lines 147+148 my brain says "index is 1 too high" b/c it thinks `copy_bytes` is the capacity.

Reply:
Done